ddtrace/internal/runtime/__init__.py

```diff
@@ -9,4 +12,50 @@
     "RuntimeTags",
     "RuntimeMetrics",
     "RuntimeWorker",
+    "get_runtime_id",
 ]
+
+
+def _generate_runtime_id():
+    return uuid.uuid4().hex
+
+
+_RUNTIME_ID = _generate_runtime_id()
+
+
+if hasattr(os, "register_at_fork"):
```

Comment:
I'm most definetely not a Python developer, so feel free to ignore.
Fee things you may want to consider here:

* Using 'double check' or any sort of optimistic locking strategy may help to avoid global lock.
* There is https://github.com/google/python-atfork that seems to work before 3.7 - in fact I guess on any reasonably python and POSIX OS.
* Both 'register_at_fork' and 'atfork' mentioned above seems to only work if forked from python code - not sure how much this is a concern.

Reply:
There is also https://github.com/vfreex/pyatfork - which actually seems to be using OS functionality intended for this sort of thing.

Reply:
Yeah, it's always a burden to add external dependencies so I took the road of having two implementations — which is anyway mandatory if we want to support non-POSIX OS.

We might also be able to write a little C/Cython wrapper if we want to expose `pthread_atfork`. That's something on my plate already as we use it in the profiler and I'd love to it work for Python < 3.7.